bql/semantic/expression_test.go

```diff
@@ -496,20 +592,150 @@ func TestNewEvaluator(t *testing.T) {
+				"?foo": &table.Cell{N: newNodeFromStringOrDie("/_", "meowth")},
+			},
+			err:  false,
+			want: true,
+		},
+		{
+			id: `?o > "37"^^type:int64`,
+			in: []ConsumedElement{
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemBinding,
+					Text: "?o",
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemGT,
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemLiteral,
+					Text: `"37"^^type:int64`,
+				}),
+			},
+			r: table.Row{
+				"?o": &table.Cell{N: newNodeFromStringOrDie("/u", "paul")},
+			},
+			err:  false,
+			want: false,
+		},
+		{
+			id: `?o = /u<peter>`,
+			in: []ConsumedElement{
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemBinding,
+					Text: "?o",
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemEQ,
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemNode,
+					Text: "/u<peter>",
+				}),
+			},
+			r: table.Row{
+				"?o": &table.Cell{L: buildLiteralOrDie(`"73"^^type:int64`)},
+			},
+			err:  false,
+			want: false,
+		},
+		{
+			id: `?s ID ?id > "37"^^type:int64`,
+			in: []ConsumedElement{
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemBinding,
+					Text: "?id",
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemGT,
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemLiteral,
+					Text: `"37"^^type:int64`,
+				}),
+			},
+			r: table.Row{
+				"?id": &table.Cell{S: table.CellString("peter")},
+			},
+			err:  true,
+			want: false,
+		},
+		{
+			id: `?s ID ?id = /u<peter>`,
+			in: []ConsumedElement{
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemBinding,
+					Text: "?id",
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemEQ,
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemNode,
+					Text: "/u<peter>",
+				}),
+			},
+			r: table.Row{
+				"?id": &table.Cell{S: table.CellString("peter")},
+			},
+			err:  true,
+			want: false,
+		},
+		{
+			id: `?foo = "10"^^type:text`,
+			in: []ConsumedElement{
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemBinding,
+					Text: "?foo",
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemEQ,
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemLiteral,
+					Text: `"10"^^type:text`,
+				}),
+			},
+			r: table.Row{
+				"?foo": &table.Cell{L: buildLiteralOrDie(`"10"^^type:int64`)},
+			},
+			err:  false,
+			want: false,
+		},
+		{
+			id: `?foo > "10"^^type:int64`,
+			in: []ConsumedElement{
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemBinding,
+					Text: "?foo",
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemGT,
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemLiteral,
+					Text: `"10"^^type:int64`,
+				}),
+			},
+			r: table.Row{
+				"?foo": &table.Cell{L: buildLiteralOrDie(`"100.0"^^type:float64`)},
 			},
 			err:  false,
 			want: true,
 		},
 	}
 	for _, entry := range testTable {
```

Comment:
You don't have a lot of setup to do in this test. I'd advise you split the test function into two to avoid the complexity you add here with all the ifs.

Reply:
I will be doing this in the next PR. Thank you once again Tati.